RUFAS/biophysical/manure/storage/composting.py

```diff
@@ -0,0 +1,574 @@
+    def _calculate_dry_matter_loss(methane_emission: float, carbon_decomposition: float) -> float:
+        """
+        This function calculates the total dry matter loss of the current day.
+
+        Returns
+        -------
+        float
+            The total dry matter loss of the current day, kg/day.
+        """
+        return 2 * carbon_decomposition + methane_emission
+
+    @staticmethod
+    def _calculate_carbon_decomposition(
+        manure_temperature: float, non_degradable_volatile_solids: float, total_solids: float
+    ) -> float:
+        """
+        This function calculates the total carbon decomposition of the current day.
+
+        Parameters
+        ----------
+        manure_temperature : float
+            The manure temperature of the current day, Celsius.
+        non_degradable_volatile_solids : float
+            The non-degradable volatile solids of the current day, kg.
+        total_solids : float
+            The total solids of the current day, kg.
+
+        Returns
+        -------
+        float
+            The total carbon decomposition of the current day, kg/day.
+        """
+        carbon_decomposition_rate = Composting._calculate_carbon_decomposition_rate(manure_temperature)
+        anaerobic_coefficient = Composting._calculate_anaerobic_coefficient()
+
+        return (
+            (
+                total_solids * DEFAULT_CARBON_FRACTION_AVAILABLE_IN_MANURE
+                + non_degradable_volatile_solids * DEFAULT_CARBON_FRACTION_AVAILABLE_IN_BEDDING
+            )
+            * carbon_decomposition_rate
+            * DEFAULT_EFFECT_OF_MOISTURE_ON_MICROBIAL_DECOMPOSITION
+            * anaerobic_coefficient
+        )
+
+    @staticmethod
+    def _calculate_carbon_decomposition_rate(manure_temperature: float) -> float:
+        """
+        This function calculates the carbon decomposition rate of the current day.
+
+        Parameters
+        ----------
+        manure_temperature : float
+            The manure temperature of the current day, Celsius.
+
+        Returns
+        -------
+        float
+            The carbon decomposition rate of the current day, per day.
+        """
+        max_microbial_decomposition_rate = Composting._calculate_max_microbial_decomposition_rate()
+        slow_microbial_decomposition_rate = Composting._calculate_slow_microbial_decomposition_rate(manure_temperature)
+
+        return float(
+            (
+                (max_microbial_decomposition_rate - slow_microbial_decomposition_rate)
+                * (
+                    math.e
+                    ** (DEFAULT_FIRST_ORDER_DECAYING_COEFFICIENT * (DEFAULT_DAYS_SINCE_LAST_TILLAGE - DEFAULT_LAG_TIME))
+                )
+                * slow_microbial_decomposition_rate
+            )
+        )
+
+    @staticmethod
+    def _calculate_max_microbial_decomposition_rate() -> float:
+        """
+        This function calculates the max microbial decomposition rate of the current day.
```

Comment:
```suggestion
        This function calculates the max microbial decomposition rate. This parameter is set to 0.04195 but the equation and set values are shown below for reference.
```
Should the 'current day' specification be removed because this value is static, e.g. does not vary across days?

Reply:
I think that makes sense to remove the `current day` part for clarity.